maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java

```diff
@@ -1200,7 +1200,7 @@ public ModelSource getModelSource()
                 }
                 else
                 {
-                    if ( childModel.getVersion().contains( "${" ) )
+                    if ( childModel.getVersion().contains( "${project.version}" ) )
```

Comment:
Shouldn't it be equals?

Reply:
Should be and I should have committed the test fix the dang test (which I am doing now) sorry!

Reply:
Update:  So the funny thing is the thing we are trying to avoid by checking the version for property replacement *doesn't* work with the most likely case one would see in the wild (which is why I *can't* get the test to throw properly).

When the test is actually executed with the below pom the `childModel.getVersion()` returns `1` not `${project.version}`!

```
<project>
  <modelVersion>4.0.0</modelVersion>
  <parent>
    <groupId>org.apache</groupId>
    <artifactId>apache</artifactId>
    <version>[1,1]</version>
  </parent>
  <artifactId>child</artifactId>
  <!-- Must not be project.version from parent due to version range. -->
  <version>${project.version}</version>
  <packaging>pom</packaging>
</project>

```

Reply:
I believe I have handled both cases and properly updated the tests.